include/deal.II/fe/fe_poly_tensor.h

```diff
@@ -247,11 +252,30 @@ class FE_PolyTensor : public FiniteElement<dim, spacedim>
     // initialize fields only if really
     // necessary. otherwise, don't
     // allocate memory
+
+    const bool update_transformed_shape_values =
+      (std::find_if(this->mapping_type.begin(),
+                    this->mapping_type.end(),
+                    [](const MappingType t) { return t != mapping_none; }) !=
+       this->mapping_type.end());
```

Comment:
I didn't know about those! Should I use `std::any_of` since updating needs to happen if any of the mapping types are nontrivial?

I applied the same change to the `std::find_if` below this as well.

Reply:
Nice -- I had no idea about `std::any_of` either -- which is why I had suggested the `std::find_if` version above to @GrahamBenHarper :-)